File: Algorithms/Python3.x/824-Goat_Latin.py
# Runtime: 16 ms, faster than 99.93% of Python3 online submissions for Goat Latin.
# Memory Usage: 14.1 MB, less than 90.41% of Python3 online submissions for Goat Latin.
class Solution:
    def toGoatLatin(self, sentence: str) -> str:
        words = sentence.split(' ')
        res = ""
        for i, word in enumerate(words):
            res += (word if word[0] in 'aeiouAEIOU' else word[1:] + word[0]) + 'ma'
            res += (i + 1) * 'a' + ' '
        return res[:-1]

File: Algorithms/C++/824-Goat_Latin.cpp
// Runtime: 4 ms, faster than 48.53% of C++ online submissions for Goat Latin.
// Memory Usage: 6.7 MB, less than 33.37% of C++ online submissions for Goat Latin.

class Solution {
public:
    string toGoatLatin(string S) {
        std::string res, word, vowel("aeiouAEIOU");
        int a_cnt = 1, i = 0;
        while (i <= S.size()) {
            if (S[i] == ' ' || i == S.size()) {
                if (vowel.find(word[0]) != std::string::npos) {
                    res += word;
                } else {
                    res += word.substr(1, word.size() - 1) + word[0];
                }
                res += "ma" + std::string(a_cnt, 'a') + " ";
                a_cnt++;
                word = "";
            } else {
                word += S[i];
            }
            i++;
        }
        return res.substr(0, res.size() - 1);
    }
};
